demonstrations/tutorial_zne_catalyst.py

```diff
@@ -0,0 +1,279 @@
+r"""
+Digital zero-noise extrapolation with Catalyst
+==============================================
+
+In this tutorial, you will learn how to use error mitigation, and in particular 
+the zero-noise extrapolation (ZNE) technique, in combination with 
+`Catalyst <https://docs.pennylane.ai/projects/catalyst>`_, a framework for quantum 
+just-in-time (JIT) compilation with PennyLane. 
+We'll demonstrate how to generate noise-scaled circuits, execute them on a noisy quantum
+simulator, and use extrapolation techniques to estimate the zero-noise result, all while
+leveraging JIT compilation through Catalyst.
+
+Using ZNE with PennyLane
+------------------------
+
+The demo :doc:`Error mitigation with Mitiq and PennyLane <tutorial_error_mitigation>`
+shows how ZNE, along with other error mitigation techniques, can be carried out in PennyLane
+by using `Mitiq <https://github.com/unitaryfund/mitiq>`__, a Python library developed by `Unitary Fund <https://unitary.fund/>`__.
+
+ZNE in particular is also offered out of the box in PennyLane as a *differentiable* error mitigation technique,
+for usage in combination with variational workflows. More on this in the tutorial
+:doc:`Differentiating quantum error mitigation transforms <tutorial_diffable-mitigation>`.
+
+On top of the error mitigation routines offered in PennyLane, ZNE is also available for just-in-time
+(JIT) compilation. In this tutorial we see how an error mitigation routine can be
+integrated in a Catalyst workflow.
+
+At the end of the tutorial, we will compare time for the execution of ZNE routines in
+pure PennyLane vs. PennyLane Catalyst with JIT.
```

Comment:
```suggestion
At the end of the tutorial, we will compare the execution time of ZNE routines in
pure PennyLane vs. PennyLane and Catalyst with JIT.
```
For example, this would be more correct.